src/main/java/uk/nhs/digital/nhsconnect/nhais/mesh/http/MeshConfig.java

```diff
@@ -34,18 +41,37 @@ public MeshConfig(
     }
 
     public String getEndpointCert() {
-        String cert = endpointCert;  //below computations are needed when default certificate is imported from application.yml
-        cert = cert.replaceAll("-----BEGIN CERTIFICATE-----", "");
-        cert = cert.replaceAll("-----END CERTIFICATE-----", "");
-        cert = cert.replaceAll(" ", "\n");
-        return "-----BEGIN CERTIFICATE-----\n" + cert + "-----END CERTIFICATE-----";
+        return trimExtraWhitespace(endpointCert);
     }
 
     public String getEndpointPrivateKey() {
-        String key = endpointPrivateKey; //below computations are needed when default private key is imported from application.yml
-        key = key.replaceAll("-----BEGIN RSA PRIVATE KEY-----", "");
-        key = key.replaceAll("-----END RSA PRIVATE KEY-----", "");
-        key = key.replaceAll(" ", "\n");
-        return "-----BEGIN RSA PRIVATE KEY-----\n" + key + "-----END RSA PRIVATE KEY-----";
+        return trimExtraWhitespace(endpointPrivateKey);
+    }
+
+    /**
+     * Different methods of importing the certificates (application.yml, ENV, Cloud secret) can affect whitespace
+     * and line delimiters. For these to be read as valid PEM files the whitespace needs to be stripped and newlines
+     * included appropriately. This method parses and reformats these inputs into strings that can be read as PEM files.
+     *
+     * @param value the certificate or key to reform
+     * @return the reformatted certificate or key
+     */
+    private String trimExtraWhitespace(String value) {
```

Comment:
I'd move this method to separate class - what it does don't belong to `MeshConfig`

Reply:
Also `formatCertificate` might be a better name, as this method does more than trimming